experimental/sql/test/impl_to_iterators/map.xdsl

```diff
@@ -20,6 +20,7 @@ module() {
 // CHECK-NEXT:     ^0(%{{.*}} : !llvm.struct<"", [!i32]>):
 // CHECK-NEXT:       %{{.*}} : !i32 = llvm.extractvalue(%{{.*}} : !llvm.struct<"", [!i32]>) ["position" = [0 : !index]]
 // CHECK-NEXT:       %{{.*}} : !i32 = arith.addi(%{{.*}} : !i32, %{{.*}} : !i32)
+// CHECK-NEXT:       %{{.*}} : !llvm.struct<"", [!i32]> = llvm.mlir.undef()
 // CHECK-NEXT:       %{{.*}} : !llvm.struct<"", [!i32]> = llvm.insertvalue(%{{.*}} : !llvm.struct<"", [!i32]>, %{{.*}} : !i32) ["position" = [0 : !index]]
```

Comment:
I am not an expert with `FileCheck` and much less one on best practices with it but it seems odd like the line with `insertvalue` has not changed. I think that this is because you only use wildcards (`{{.*}}`) and not variable binding. With the latter, you'd have `%[[V1.*]] : ... llvm.mlir.undef()` to bind the name of the SSA value to `V1` and then `... llvm.insertvalue(%[[V1] : ...) ...` to match that name in subsequent lines. This way you can check whether `insertvalue` uses the correct operands. The downside, of course, is that it takes (even) longer to write these tests...

If you do want to change your tests to the stricter form, don't do it all at once. Just apply the new style gradually over time...

Reply:
I think you are right, I should probably restructure this, since it does matter here, whether we return the result of the `llvm.mlir.undef` or the result of the `insertvalue`.

Reply:
It seems that filecheck (a python reimplementation of FileCheck, that does not require compiling llvm) cannot handle these kinds of matchings. I added this style of testing to the `test_mlir` folder, that contains other tests that would require building LLVM, but are not yet supported by our CI.

I also found a small issue by testing like this, so that was worth it :)